account_invoice_import/wizard/account_invoice_import.py

```diff
@@ -0,0 +1,1448 @@
+                        break
+            if not company_id:
+                logger.error(
+                    "Invoice import mail gateway in multi-company setup: "
+                    "invoice_import_email of the companies of this DB was "
+                    "not found as destination of this email (to: %s, cc: %s). "
+                    "Ignoring this email.",
+                    msg_dict["email_to"],
+                    msg_dict["cc"],
+                )
+                return
+        else:  # mono-company setup
+            company_id = all_companies[0]["id"]
+
+        self = self.with_company(company_id)
+        i = 0
+        if msg_dict.get("attachments"):
+            i += 1
+            for attach in msg_dict["attachments"]:
+                logger.info(
+                    "Attachment %d: %s. Trying to import it as an invoice",
+                    i,
+                    attach.fname,
+                )
+                origin = _("email sent by <b>%s</b> on %s with subject <i>%s</i>") % (
+                    msg_dict.get("email_from"),
+                    msg_dict.get("date"),
+                    msg_dict.get("subject"),
+                )
+                try:
+                    invoice_id = self.create_invoice_webservice(
+                        base64.b64encode(attach.content),
+                        attach.fname,
+                        origin,
+                        company_id=company_id,
+                    )
+                    logger.info(
+                        "Invoice ID %d created from email attachment %s.",
+                        invoice_id,
+                        attach.fname,
+                    )
+                except Exception as e:
+                    logger.error(
+                        "Failed to import invoice from mail attachment %s. Error: %s",
+                        attach.fname,
+                        e,
+                    )
+        else:
+            logger.info("The email has no attachments, skipped.")
+        # message_new() is normally used on a models.Model that inherit from
+        # mail.thread and it should return the recordset of the model it creates
+        # As we return None, we get this error in the logs:
+        # Traceback (most recent call last):
+        #  File "xx/mail/models/mail_thread.py", line 1050, in _message_route_process
+        #  subtype_id = thread._creation_subtype().id
+        # AttributeError:
+        # 'account.invoice.import' object has no attribute '_creation_subtype'
+        # This traceback doesn't block the creation of the invoice.
+        # I could move this code to account.move, but I don't want to
+        # create an invoice on every email received, I still want to return
+        # nothing when there is no attachment or when the attachment
+        # fails to import or when the invoice is already created
+        # So, for the moment, we leave it like that...
+        # If you have a better idea, please suggest!
```

Comment:
I've implemented the obvious solution here: https://github.com/akretion/edi/pull/4, letting this model inherit from mail.thread, and it works. Maybe you have considered it before, and there may be good reasons not to use it that I've missed?

Reply:
Thanks !

Reply:
Thanks for taking it! I see now that I did not remove this comment in the proposed PR. Can you squash that in somewhere?

Reply:
And a rebase while you are at it, to squash out the merge commit from my intermediate PR